mtools/util/logfile.py

```diff
@@ -206,6 +214,41 @@ def storage_engine(self):
             self._iterate_lines()
         return self._storage_engine
 
+    @property
+    def shards(self):
+        """Return the shards (if available)"""
+        if not self._num_lines:
+            self._iterate_lines()
```

Comment:
The sharding details should ideally be in a separate lazy method rather than `_iterate_lines()`, so extra matches are only run when needed.

Reply:
Created a seperate lazy method to find all the sharding information